### DISCUSSION TITLE
Cursor tooltip stealing focus

### DISCUSSION THREAD

**Discussion by mushroomtaste** [whs/runekit#10] [Q&A Linux]
Hey, loving being able to use use this on Linux

Only real problem I have is that whenever I use cursor tooltips with AFKwarden on Linux Mint Cinnamon they steal focus away from the RS3 window which makes them frustrating to use. Which is a shame due to them being my favourite alarm type when I used Windows OS.

I tried using the "windows" configs to change how window focus works to no avail.

Thanks

**Comment by schoentoon** [comment]
I'll install Cinnamon on my Ubuntu 20.04 system and try to reproduce this to figure out where this is coming from. What version of Linux Mint are you on if you don't mind me asking.

**Reply by mushroomtaste** [reply]
Linux Mint 20.1 Cinnamon
Cinnamon version: 4.8.6

Can't close runekit if a cursor notification gets stuck. Have to restart RS3 to get it to reset.

**Reply by schoentoon** [reply]
I tried it just now and I couldn't get it to reproduce. I assume that with you as well the popup actually shows up in the notification tray, rather than at the mouse cursor?
Also capturing related alerts (like prayer etc) didn't seem to work out of the box for me somehow.

**Reply by mushroomtaste** [reply]
I use the interface options in main toolkit settings, notification works fine it replaces "show cursor tooltip" in individual option in AFKwarden with a regular notification with the text entered. 

If I now change it to cursor tooltips in main Runekit settings it changes it to just a cursor tooltip which will require clicking once on RS3 to remove the tooltip following and blocking the cursor and then can click on RS3 as normal.

Also showing progress bar on bottom window bar doesn't work.

I don't know of any way to try cursor tooltips besides Runekit on my system so I can't find out if its just a problem with my system.

I've had programming and other technical stuff be weird and fix n' break over the years with no logical sense so maybe its just a fluke.

Edit: Tried redownloading appimage from here and its still the same https://github.com/whs/runekit/releases/tag/continuous

**Reply by schoentoon** [reply]
Oh my bad, I have that set to notification rather than cursor tooltip. I'll again after changing that, although that's something for tomorrow most likely.

**Reply by mushroomtaste** [reply]
Coolio, thanks for checking it out

**Reply by schoentoon** [reply]
I found a quick minute to try it again. Also found out that I have to launch runekit after launching RuneScape for it to actually capture anything, so prayer alert etc did work. I can see what you mean with it stealing focus. And it does seem Cinnamon specify, as it acts differently (as in it doesn't seem to steal focus) on xfce4 which I usually use. Will investigate further at a later point.

**Reply by whs** [reply]
I'm guessing the window manager wants to focus any new window by default, which we don't want. Maybe look into raising the focus stealing prevention level? Or I think some WM might have that focus new window option that needs to be disabled.

If you're looking for code level changes, I'm guessing that moving the tooltip offscreen while it is "hidden" instead of actually removing the window might help. Not sure about side effects though like it appearing in alt-tab or "gather window" might force move it.

**Reply by schoentoon** [reply]
Apparently didn't even test this properly on xfce4. It seems to steal focus there as well I found out today by accident
